File: cricket-team-management/src/main/java/com/gorl/demo/service/PlayerService.java
package com.gorl.demo.service;

import com.gorl.demo.dto.PlayerDTO;

import java.util.List;

public interface PlayerService {

    List<PlayerDTO> getAllPlayers();

    PlayerDTO getPlayerById(Long playerId);

    PlayerDTO createPlayer(PlayerDTO playerDTO);

    PlayerDTO updatePlayer(Long playerId, PlayerDTO playerDTO);

    void deletePlayer(Long playerId);
}

File: src/services/playerService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api/players';

const api = axios.create({
  baseURL: API_BASE_URL,
  headers: { 'Content-Type': 'application/json' },
});

const playerService = {
  getAllPlayers: async () => {
    const response = await api.get('');
    return response.data;
  },

  getPlayerById: async (playerId) => {
    const response = await api.get(`/${playerId}`);
    return response.data;
  },

  createPlayer: async (playerData) => {
    const response = await api.post('', playerData);
    return response.data;
  },

  updatePlayer: async (playerId, playerData) => {
    const response = await api.put(`/${playerId}`, playerData);
    return response.data;
  },

  deletePlayer: async (playerId) => {
    const response = await api.delete(`/${playerId}`);
    return response.data;
  },
};

export default playerService;